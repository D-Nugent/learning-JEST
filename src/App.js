import React,{useState} from 'react';
import './App.css';

function App() {
  const [count,setCount] = useState(0);

  const handleIncrementClick = () => {
    setCount(prevCount => (prevCount+1))
  }

  const handleResetClick = () => {
    setCount(0);
  }

  return (
    <div className="App">
      <p>Count: {count}</p>
      <button className="increment" onClick={handleIncrementClick}>Increment Count</button>
      <button className="reset" onClick={handleResetClick}>Reset Count</button>
    </div>
  );
}

export default App;
