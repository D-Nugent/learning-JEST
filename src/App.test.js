import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import App from './App';

describe('App Component', () => {
  it('starts with a count of 0', () => {
    const wrapper = shallow(<App />);
    const text = wrapper.find('p').text();
    expect(text).toEqual('Count: 0')
  });
  it ('increments count by 1 when increment button is clicked', () => {
    const wrapper = shallow(<App />);
    const incrBtn = wrapper.find('button.increment');
    incrBtn.simulate('click');
    const text = wrapper.find('p').text();
    expect(text).toEqual('Count: 1')
  });
  describe('resets count to zero', () => {
    beforeEach(() => {
      const wrapper = shallow(<App />);
      const incrBtn = wrapper.find('button.increment');
      incrBtn.simulate('click');
      const text = wrapper.find('p').text();
      expect(text).toEqual('Count: 1')
    })
    
    it ('with the count set at 1, reset returns to 0', () => {
      const wrapper = shallow(<App />);
      const resetBtn = wrapper.find('button.reset');
      resetBtn.simulate('click')
      const text = wrapper.find('p').text();
      expect(text).toEqual('Count: 0')
    })
  })
  it('matches the snapshot', () => {
    const tree = create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})

describe('Basic math', () => {
  it('knows 2+2=4', () => {
    expect(2+2).toBe(4)
  })
})