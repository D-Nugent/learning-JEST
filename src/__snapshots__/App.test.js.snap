// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches the snapshot 1`] = `
<div
  className="App"
>
  <p>
    Count: 
    0
  </p>
  <button
    className="increment"
    onClick={[Function]}
  >
    Increment Count
  </button>
  <button
    className="reset"
    onClick={[Function]}
  >
    Reset Count
  </button>
</div>
`;
